eogrow/tasks/prediction.py

```diff
@@ -81,8 +83,11 @@ def apply_predictor(
             return return_on_empty
 
         if self.mp_lock:
-            return execute_with_mp_lock(predictor, processed_features)
-        return predictor(processed_features)
+            predictions = execute_with_mp_lock(predictor, processed_features)
+        else:
+            predictions = predictor(processed_features)
+
+        return predictions.astype(self.output_dtype) if self.output_dtype else predictions
```

Comment:
What if the default is that the same `dtype` as input features? Might make more sense.

Reply:
what about classification?

Reply:
Ah, good point... Then I it's most elegant the way it is now.

Reply:
Is there any community preference in such cases between

```python
return predictions.astype(self.output_dtype) if self.output_dtype else predictions
```

and 

```python
if self.output_dtype:
    predictions = predictions.astype(self.output_dtype)

return predictions
```

I can see pros/cons in both, but not sure if this is just personal preference or not.

Reply:
I prefer the first one since it's expression based (and takes up less space), but i'd say it's personal preference